lib/stack_master/commands/apply.rb

```diff
@@ -62,12 +62,16 @@ def create_or_update_stack
       end
 
       def create_stack
+        upload_files
+        options = stack_options.merge(tags: proposed_stack.aws_tags, change_set_type: 'CREATE')
+        @change_set = ChangeSet.create(options)
+        halt!(@change_set.status_reason) if @change_set.failed?
+        @change_set.display(StackMaster.stdout)
         unless ask?('Create stack (y/n)? ')
-          failed!("Stack creation aborted")
+          cf.delete_stack(stack_name: stack_name)
+          halt! 'Stack creation aborted'
         end
-        upload_files
-        on_failure = @config.stack_defaults['on_failure'] || @options.on_failure
-        cf.create_stack(stack_options.merge({tags: proposed_stack.aws_tags, on_failure: on_failure}))
```

Comment:
This is removing existing functionality that allows you to specify an `on_failure` option when creating a stack. This is useful when you want the stack to be deleted if it fails to create the first time.

I wonder if we can still allow this by continuing to use `create_stack` if the `on_failure` option is explicitly provided (as it will just Rollback by default) and provide some documentation to indicate you can't use that option with SAM.